src/unix/stream.c

```diff
@@ -301,7 +301,9 @@ int uv__stream_try_select(uv_stream_t* stream, int* fd) {
   if (fds[1] > max_fd)
     max_fd = fds[1];
 
-  sread_sz = (max_fd + NBBY) / NBBY;
+  sread_sz = (max_fd + 1) / NBBY;
+  if (sread_sz % sizeof(uint32_t) != 0)
+    sread_sz += sizeof(uint32_t) - (sread_sz % sizeof(uint32_t));
```

Comment:
Ben's suggestion sounds good, mind doing that, Ole?
On Mar 5, 2015 4:17 PM, "Ole André Vadla Ravnås" <notifications@github.com>
wrote:

> In src/unix/stream.c
> <https://github.com/libuv/libuv/pull/241#discussion_r25870434>:
>
> > @@ -301,7 +301,9 @@ int uv__stream_try_select(uv_stream_t* stream, int* fd) {
> >    if (fds[1] > max_fd)
> >      max_fd = fds[1];
> >
> > -  sread_sz = (max_fd + NBBY) / NBBY;
> > +  sread_sz = (max_fd + 1) / NBBY;
> > +  if (sread_sz % sizeof(uint32_t) != 0)
> > +    sread_sz += sizeof(uint32_t) - (sread_sz % sizeof(uint32_t));
>
> @bnoordhuis <https://github.com/bnoordhuis> [image: :+1:] (I was tempted
> to introduce a macro like that, but went with minimal changes as I'm not
> yet familiar with the libuv code-base.)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/libuv/libuv/pull/241/files#r25870434>.
>